src/main.zig

```diff
@@ -523,6 +523,8 @@ const usage_build_generic =
     \\  -fno-sanitize-thread      Disable Thread Sanitizer
     \\  -ffuzz                    Enable fuzz testing instrumentation
     \\  -fno-fuzz                 Disable fuzz testing instrumentation
+    \\  -fallow-deprecated        Allow usage of deprecated code
+    \\  -fno-allow-deprecated     Disallow usage of deprecated code
```

Comment:
I've only added these flags for `zig build` (in the build runner) and here for modules, which means that the flags can't be used directly with `zig build-exe` or `zig run`.

Reply:
I don't understand what you mean here. What you've changed here is the usage string and arg handling loop for `build-exe`, `run`, etc. Clearly you can pass this to `build-exe`; otherwise, the build system wouldn't be able to pass it. What did you mean by this?

Reply:
I'm saying that `zig build-exe foo.zig -fallow-deprecated` doesn't work, because that's a *module* flag, not a global flag. 

Maybe a global flag is desirable but then I'm not sure if it's ok for all of those to have the same name.

Reply:
That's not how this works. If you never use `-M`, there's effectively an implicit `-Mroot=the_source_file_you_passed` at the end. If per-module flags didn't work with an implicit root module, then huge chunks of the compiler usage would be broken.

Reply:
Ah I see, makes sense